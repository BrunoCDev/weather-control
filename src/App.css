.app {
  width: 100%;
  height: 100%; }

.main-container {
  width: 25%;
  border-style: solid;
  border-width: 1px;
  border-color: red; }

.title-container {
  text-align: center; }

.area {
  width: 100%;
  height: 100%; }

.content {
  width: 100%;
  display: inline-flex;
  font-size: 10px; }

.instance-container {
  width: 250px;
  height: 250px; }
