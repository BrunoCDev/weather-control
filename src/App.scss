.App {
  width: 35%;
  height: 150px;
}

.main-container {
  width: 100%;
  height: 100%;
}

.title-container {
  text-align: center;
}

.area {
  width: 100%;
  height: 100%;
  border-style: solid;
  border-width: 1px;
  border-color: red;
}

.content {
  display: inline-flex;
  font-size: 10px;
}

.instance-container {
  width: 100px;
  height: 100%;
  text-align: center;
}

.icon-container {
  width: 80%;
  height: 80%;
}

.icon {
  width: 50%;
  height: 50%;
}
