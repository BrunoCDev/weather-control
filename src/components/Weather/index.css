.weather-container {
  width: 20vw;
  height: 32vh; }
  @media (min-width: 0px) {
    .weather-container {
      height: 160px;
      width: 250px; } }
  @media (min-width: 550px) {
    .weather-container {
      height: 230px;
      width: 25%; } }
  @media (min-width: 800px) {
    .weather-container {
      height: 320px;
      width: 180px; } }

.weather-header {
  height: 3.5vh;
  width: 20vw; }
  @media (min-width: 0px) {
    .weather-header {
      height: 30px;
      width: 62px; } }
  @media (min-width: 550px) {
    .weather-header {
      height: 50px;
      width: 120px; } }
  @media (min-width: 800px) {
    .weather-header {
      height: 60px;
      width: 180px; } }

.weekday-text {
  color: #adb9c1;
  font-weight: bold;
  letter-spacing: 0.5px;
  text-align: center;
  vertical-align: middle;
  margin: 0px; }
  @media (min-width: 0px) {
    .weekday-text {
      height: 30px;
      width: 62px;
      font-size: 10px;
      line-height: 30px; } }
  @media (min-width: 550px) {
    .weekday-text {
      height: 50px;
      width: 120px;
      font-size: 15px;
      line-height: 50px; } }
  @media (min-width: 800px) {
    .weekday-text {
      height: 60px;
      width: 180px;
      font-size: 20px;
      line-height: 60px; } }

.weather-body {
  height: 27.5vh;
  width: 100%;
  text-align: center;
  vertical-align: middle; }
  @media (min-width: 0px) {
    .weather-body {
      height: 130px; } }
  @media (min-width: 550px) {
    .weather-body {
      height: 230px; } }
  @media (min-width: 800px) {
    .weather-body {
      height: 260px; } }

.icon {
  width: 15vw;
  height: auto; }
  @media (min-width: 0px) {
    .icon {
      width: 40px; } }
  @media (min-width: 550px) {
    .icon {
      width: 80px; } }
  @media (min-width: 800px) {
    .icon {
      width: 80px; } }

.description-text {
  color: #ffffff;
  width: 100%;
  height: 2vh;
  font-size: 3vw;
  letter-spacing: 0.7px; }
  @media (min-width: 0px) {
    .description-text {
      height: 12px;
      font-size: 9px; } }
  @media (min-width: 550px) {
    .description-text {
      height: 20px;
      font-size: 15px; } }
  @media (min-width: 800px) {
    .description-text {
      height: 20px;
      font-size: 22px; } }

.temp-cel {
  color: #ffffff;
  font-size: 5vw;
  letter-spacing: 1.5px; }
  @media (min-width: 0px) {
    .temp-cel {
      font-size: 15px; } }
  @media (min-width: 550px) {
    .temp-cel {
      font-size: 20px; } }
  @media (min-width: 800px) {
    .temp-cel {
      font-size: 30px; } }

.temp-fel {
  color: #ffffff;
  text-align: center;
  font-size: 3vw;
  line-height: 0px;
  letter-spacing: 0.5px; }
  @media (min-width: 0px) {
    .temp-fel {
      font-size: 9px; } }
  @media (min-width: 550px) {
    .temp-fel {
      font-size: 15px; } }
  @media (min-width: 800px) {
    .temp-fel {
      font-size: 20px; } }
