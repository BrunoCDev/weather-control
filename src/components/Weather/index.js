import React, { Component } from "react";
import "./index.css";

class Weather extends Component {
  constructor(props) {
    super(props);
    this.state = {
      week: [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday"
      ],
      loading: true
    };
    this.changeColor = this.changeColor.bind(this);
    this.parseDescription = this.parseDescription.bind(this);
    this.parseTemp = this.parseTemp.bind(this);
  }

  componentDidMount() {
    this.setState({ loading: false });
  }

  // Changes the <div> color to create pattern.
  changeColor(i, str) {
    if (i % 2 === 0 && str === "header") {
      return "#282936";
    }
    if (i % 2 !== 0 && str === "header") {
      return "#1F212B";
    }
    if (i % 2 === 0 && str === "body") {
      return "#2C2E3C";
    } else {
      return "#222530";
    }
  }

  // Uppercases the first letter of the description.
  parseDescription(description) {
    description = description.split(" ");
    const result = [];
    description.map(el => {
      result.push(el[0].toUpperCase() + el.slice(1, el.length));
    });
    return result.join(" ");
  }

  // Converts temperature received by API (Kelvin) to Celsius or Fahrenheit.
  parseTemp(temp, str) {
    if (str === "C") {
      return `${Math.round(temp - 273.15)}°C`;
    }
    if (str === "F") {
      return `${Math.round((temp - 273.15) * 1.8 + 32)}°F`;
    }
  }

  render() {
    // Object destructuring for shorthand.
    const data = this.props.data;
    const { icon, description } = data.info.weather[0];
    const { temp } = data.info.main;
    const { week, loading } = this.state;
    const { index } = this.props;

    // Gets the day number.
    const date = data.info.dt_txt.split(" ");
    const day = date[0].split("-");

    // Conditional Render so the component changes the displayed information after
    // the information changes.
    return (
      !loading && (
        <div
          className="weather-container"
          style={{ backgroundColor: this.changeColor(index, "header") }}
        >
          <div
            className="weather-header"
            style={{ backgroundColor: this.changeColor(index, "header") }}
          >
            <p className="weekday-text">
              {index === 0
                ? "Today " + day[2]
                : week[data.weekDay] + ` ${day[2]}`}
            </p>
          </div>
          <div
            className="weather-body"
            style={{ backgroundColor: this.changeColor(index, "body") }}
          >
            <div>
              <img
                className="icon"
                src={`http://openweathermap.org/img/w/${icon}.png`}
                alt="Weather"
              />
            </div>
            <div>
              <p className="description-text">
                {this.parseDescription(description)}
              </p>
            </div>
            <div>
              <p className="temp-cel">{this.parseTemp(temp, "C")}</p>
            </div>
            <div className="temp-fel-container">
              <p className="temp-fel">{this.parseTemp(temp, "F")}</p>
            </div>
          </div>
        </div>
      )
    );
  }
}

export default Weather;
