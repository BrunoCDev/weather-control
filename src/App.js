import React, { Component } from "react";
import "./App.css";
import axios from "axios";
import ScrollArea from "react-scrollbar";
import moment from "moment";

class App extends Component {
  constructor() {
    super();
    this.state = {
      city: "London",
      date: {},
      loading: true,
      week: [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday"
      ]
    };
    this.parseWeatherInformation = this.parseWeatherInformation.bind(this);
  }

  componentDidMount() {
    axios
      .post("http://localhost:3005/api/get/london", { city: this.state.city })
      .then(response => {
        const { data } = response;
        this.parseWeatherInformation(data);
        this.setState({
          city: data.city,
          loading: false
        });
      })
      .catch(console.log);
  }

  parseWeatherInformation(data) {
    data.list.map((info, i) => {
      if (data.list.length - 1 !== i) {
        const date = info.dt_txt.split(" ");
        if (i === 0) {
          this.setState({
            date: [{ info, weekDay: moment(date[0]).day() }]
          });
        } else {
          if (this.state.date.length !== 4) {
            const lastDate = this.state.date[0].info.dt_txt.split(" ");
            if (date[1] === lastDate[1]) {
              this.setState(prevState => ({
                date: [
                  ...prevState.date,
                  { info, weekDay: moment(date[0]).day() }
                ]
              }));
            }
          }
        }
      }
    });
  }

  render() {
    const { city, week } = this.state;
    console.log(this.state);
    return (
      <div className="App">
        {!this.state.loading ? (
          <div className="main-container">
            <div className="title-container">
              <strong className="title">
                {city.name}, {city.country}
              </strong>
            </div>
            <ScrollArea
              className="area"
              contentClassName="content"
              speed={0.8}
              vertical={false}
            >
              {this.state.date.map((info, i) => {
                const el = info.info;
                const date = el.dt_txt.split(" ");
                const time = date[1].slice(0, 5);
                return (
                  <div key={i} className="instance-container">
                    <p className="date">{week[info.weekDay]}</p>
                    <p className="time"> {time}</p>
                    <img
                      className="icon"
                      src={`http://openweathermap.org/img/w/${
                        el.weather[0].icon
                      }.png`}
                      alt={el.weather[0].main}
                    />
                    <p className="description">{el.weather[0].description}</p>
                    <p className="temp">
                      {Math.round(el.main.temp - 273.15)}Â°C{" "}
                    </p>
                  </div>
                );
              })}
            </ScrollArea>
          </div>
        ) : null}
      </div>
    );
  }
}

export default App;
