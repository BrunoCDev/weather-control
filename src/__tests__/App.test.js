import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

import Weather from "./../components/Weather";
import App from "./../App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  const renderedComponent = ReactDOM.render(<App />, div);
  expect(renderedComponent).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

describe("<App/>", () => {
  it("Should render Weather component", () => {
    const children = <Weather />;
    const renderedComponent = shallow(<div>{children}</div>);
    expect(renderedComponent.contains(children)).toEqual(true);
  });

  it("Should adopt the className", () => {
    const renderedComponent = shallow(<div className="App" />);
    expect(renderedComponent.find("div").hasClass("App")).toEqual(true);
  });
});
