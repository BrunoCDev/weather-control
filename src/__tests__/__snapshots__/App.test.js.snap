// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
App {
  "_reactInternalFiber": FiberNode {
    "_debugID": 4,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
      "lineNumber": 15,
    },
    "alternate": FiberNode {
      "_debugID": 4,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": Object {
        "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
        "lineNumber": 15,
      },
      "alternate": [Circular],
      "child": FiberNode {
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
          "lineNumber": 84,
        },
        "alternate": FiberNode {
          "_debugID": 5,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
            "lineNumber": 84,
          },
          "alternate": [Circular],
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": false,
            "className": "App",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": false,
            "className": "App",
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <div
            class="App"
/>,
          "tag": 5,
          "type": "div",
          "updateQueue": null,
        },
        "child": null,
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": false,
          "className": "App",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": false,
          "className": "App",
        },
        "ref": null,
        "return": [Circular],
        "sibling": null,
        "stateNode": <div
          class="App"
/>,
        "tag": 5,
        "type": "div",
        "updateQueue": null,
      },
      "effectTag": 1,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {},
      "memoizedState": Object {
        "city": "",
        "citySearch": "",
        "data": Array [],
        "loading": true,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {},
      "ref": null,
      "return": FiberNode {
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": FiberNode {
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": [Circular],
          "child": [Circular],
          "effectTag": 32,
          "expirationTime": 1,
          "firstEffect": [Circular],
          "index": 0,
          "key": null,
          "lastEffect": [Circular],
          "memoizedProps": null,
          "memoizedState": Object {
            "element": <App />,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "sibling": null,
          "stateNode": Object {
            "containerInfo": <div>
              <div
                            class="App"
              />
</div>,
            "context": Object {},
            "current": [Circular],
            "finishedWork": null,
            "firstBatch": null,
            "hydrate": false,
            "nextScheduledRoot": null,
            "pendingChildren": null,
            "pendingCommitExpirationTime": 0,
            "pendingContext": null,
            "remainingExpirationTime": 0,
          },
          "tag": 3,
          "type": null,
          "updateQueue": Object {
            "baseState": Object {
              "element": <App />,
            },
            "callbackList": null,
            "capturedValues": null,
            "expirationTime": 0,
            "first": null,
            "hasForceUpdate": false,
            "isInitialized": true,
            "isProcessing": false,
            "last": null,
          },
        },
        "child": [Circular],
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <App />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
                        class="App"
            />
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": null,
      },
      "sibling": null,
      "stateNode": [Circular],
      "tag": 2,
      "type": [Function],
      "updateQueue": null,
    },
    "child": FiberNode {
      "_debugID": 5,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
        "lineNumber": 84,
      },
      "alternate": FiberNode {
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": Object {
            "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
            "lineNumber": 15,
          },
          "alternate": [Circular],
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {},
          "memoizedState": Object {
            "city": "",
            "citySearch": "",
            "data": Array [],
            "loading": true,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {},
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": [Circular],
              "effectTag": 32,
              "expirationTime": 1,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <App />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="App"
                  />
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <App />,
                },
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 0,
                "first": null,
                "hasForceUpdate": false,
                "isInitialized": true,
                "isProcessing": false,
                "last": null,
              },
            },
            "child": [Circular],
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <App />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                                class="App"
                />
</div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "_debugSource": Object {
          "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
          "lineNumber": 84,
        },
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": false,
          "className": "App",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": false,
          "className": "App",
        },
        "ref": null,
        "return": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": Object {
            "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
            "lineNumber": 15,
          },
          "alternate": [Circular],
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {},
          "memoizedState": Object {
            "city": "",
            "citySearch": "",
            "data": Array [],
            "loading": true,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {},
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": [Circular],
              "effectTag": 32,
              "expirationTime": 1,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <App />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="App"
                  />
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <App />,
                },
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 0,
                "first": null,
                "hasForceUpdate": false,
                "isInitialized": true,
                "isProcessing": false,
                "last": null,
              },
            },
            "child": [Circular],
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <App />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                                class="App"
                />
</div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "sibling": null,
        "stateNode": <div
          class="App"
/>,
        "tag": 5,
        "type": "div",
        "updateQueue": null,
      },
      "child": null,
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": false,
        "className": "App",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": false,
        "className": "App",
      },
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <div
        class="App"
/>,
      "tag": 5,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 5,
    "expirationTime": 1,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": Object {
      "city": "",
      "citySearch": "",
      "data": Array [],
      "loading": true,
    },
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugID": 1,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": Object {
            "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
            "lineNumber": 15,
          },
          "alternate": [Circular],
          "child": FiberNode {
            "_debugID": 5,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": Object {
              "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
              "lineNumber": 84,
            },
            "alternate": FiberNode {
              "_debugID": 5,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
                "lineNumber": 84,
              },
              "alternate": [Circular],
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "children": false,
                "className": "App",
              },
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": Object {
                "children": false,
                "className": "App",
              },
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": <div
                class="App"
/>,
              "tag": 5,
              "type": "div",
              "updateQueue": null,
            },
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": false,
              "className": "App",
            },
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "children": false,
              "className": "App",
            },
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <div
              class="App"
/>,
            "tag": 5,
            "type": "div",
            "updateQueue": null,
          },
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {},
          "memoizedState": Object {
            "city": "",
            "citySearch": "",
            "data": Array [],
            "loading": true,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {},
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <App />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
                        class="App"
            />
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": null,
      },
      "child": [Circular],
      "effectTag": 32,
      "expirationTime": 1,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <App />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
                    class="App"
          />
</div>,
        "context": Object {},
        "current": FiberNode {
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": [Circular],
          "child": FiberNode {
            "_debugID": 4,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": Object {
              "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/__tests__/App.test.js",
              "lineNumber": 15,
            },
            "alternate": [Circular],
            "child": FiberNode {
              "_debugID": 5,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
                "lineNumber": 84,
              },
              "alternate": FiberNode {
                "_debugID": 5,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": [Circular],
                "_debugSource": Object {
                  "fileName": "/Users/Flavio/PersonalProjects/interview-tests/weather-control/src/App.js",
                  "lineNumber": 84,
                },
                "alternate": [Circular],
                "child": null,
                "effectTag": 0,
                "expirationTime": 0,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": false,
                  "className": "App",
                },
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": Object {
                  "children": false,
                  "className": "App",
                },
                "ref": null,
                "return": [Circular],
                "sibling": null,
                "stateNode": <div
                  class="App"
/>,
                "tag": 5,
                "type": "div",
                "updateQueue": null,
              },
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "children": false,
                "className": "App",
              },
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": Object {
                "children": false,
                "className": "App",
              },
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": <div
                class="App"
/>,
              "tag": 5,
              "type": "div",
              "updateQueue": null,
            },
            "effectTag": 1,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {},
            "memoizedState": Object {
              "city": "",
              "citySearch": "",
              "data": Array [],
              "loading": true,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {},
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": [Circular],
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": Object {
            "element": <App />,
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "sibling": null,
          "stateNode": [Circular],
          "tag": 3,
          "type": null,
          "updateQueue": null,
        },
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <App />,
        },
        "callbackList": null,
        "capturedValues": null,
        "expirationTime": 0,
        "first": null,
        "hasForceUpdate": false,
        "isInitialized": true,
        "isProcessing": false,
        "last": null,
      },
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": Object {
      "baseState": null,
      "callbackList": null,
      "capturedValues": null,
      "expirationTime": 1,
      "first": Object {
        "callback": null,
        "capturedValue": null,
        "expirationTime": 1,
        "isForced": false,
        "isReplace": false,
        "next": null,
        "partialState": Object {
          "citySearch": "",
        },
      },
      "hasForceUpdate": false,
      "isInitialized": false,
      "isProcessing": false,
      "last": Object {
        "callback": null,
        "capturedValue": null,
        "expirationTime": 1,
        "isForced": false,
        "isReplace": false,
        "next": null,
        "partialState": Object {
          "citySearch": "",
        },
      },
    },
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "getWeather": [Function],
  "parseWeatherInformation": [Function],
  "props": Object {},
  "refs": Object {},
  "state": Object {
    "city": "",
    "citySearch": "",
    "data": Array [],
    "loading": true,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
